### DISCUSSION TITLE
ETCD Timestamp Format Conversion

### DISCUSSION THREAD

**Discussion by ankitg-03** [etcd-io/etcd#14836] [Q&A]
Hi,
Could you please help to change the timestamp format as below.
Currently we get ETCD timestamps in the the following format - 2022-11-18T07:16:40.169Z but, our required format is 2022-11-18T07:15:24.722+02:00
Is there is a way to get timestamp in ETCD log in the required format 2022-11-18T07:15:24.722+02:00 ("yyyy-MM-dd'T'HH:mm:ss.SSSxxx") through some ETCD configuration?

**Comment by tjungblu** [comment]
etcd uses Zap, so you might want to ask this over at https://github.com/uber-go/zap

**Comment by ramil600** [comment]
@ankitg-03 from what I see etcd is not configured to change the default time format that is being passed in
"go.etcd.io/etcd/client/pkg/v3/logutil" package.
if you really need to change the format, then you need to change it the zap.go file:

const layout = "2006-01-02T15:04:05.999+07:00"

EncoderConfig: zapcore.EncoderConfig{
		TimeKey:        "ts",
		LevelKey:       "level",
		NameKey:        "logger",
		CallerKey:      "caller",
		MessageKey:     "msg",
		StacktraceKey:  "stacktrace",
		LineEnding:     zapcore.DefaultLineEnding,
		EncodeLevel:    zapcore.LowercaseLevelEncoder,
		**EncodeTime:     zapcore.TimeEncoderOfLayout(layout),**
		EncodeDuration: zapcore.StringDurationEncoder,
		EncodeCaller:   zapcore.ShortCallerEncoder,
	},

where you set EncodeTime as in the snippet above.
